modules/openapi-generator/src/main/resources/typescript-angular/configuration.mustache

```diff
@@ -157,4 +168,16 @@ export class {{configurationClassName}} {
             ? value()
             : value;
     }
+
+    private legacyEncodeParam(param: Param): string {
+        // This legacy implementation is here for backwards compatibility only.
+        // It is only correct for param-style 'simple'.
+        if(param.style === "simple") {
+            return encodeURIComponent(String(param.value));
+        }
```

Comment:
please also add the date handling from 

https://github.com/OpenAPITools/openapi-generator/blob/2a196ce3620aebf9e5ce0c9a17ee8e0c5365afb7/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java#L428